### PR TITLE
Remove unused Netty dynamic libraries.

### PR DESCRIPTION
**Pull request by lberki**
This saves ~2 metric megabytes in the size of the Bazel binary, bringing it to ~43 MiB.

I felt bad about saying "there is no more low-hanging fruit" and shipping DLLs for Linux...

### CODE REVIEW

File: third_party/BUILD [hunk 1]
```diff
@@ -517,14 +517,36 @@ java_import(
...
+    "//conditions:default": "*.bogusextension",
+})
+
+# Remove native libraries that are for a platform different from the one we are
+# building Bazel for
+genrule(
+    name = "filter_netty_dynamic_libs",
+    srcs = ["netty_tcnative/netty-tcnative-boringssl-static-2.0.24.Final.jar"],
+    outs = ["netty_tcnative/netty-tcnative-filtered.jar"],
+    cmd = "cp $< $@ && zip -qd $@ " + UNNECESSARY_DYNAMIC_LIBRARIES,
```
Comment (meisterT) [inline third_party/BUILD]: The zip file is non-deterministic, isn't it? In other words: does the bazel determinism test pass?

File: third_party/BUILD [hunk 2]
```diff
@@ -517,14 +517,36 @@ java_import(
...
+    # The .so file is an x86 one, so we can just remove it the CPU is not x86
+    "//src/conditions:arm": "*.so *.jnilib *.dll",
+    "//src/conditions:linux_aarch64": "*.so *.jnilib *.dll",
+    # Play it safe -- better have a big binary than a slow binary
+    # zip -d does require an argument. Supply something bogus.
+    "//conditions:default": "*.bogusextension",
+})
+
+# Remove native libraries that are for a platform different from the one we are
+# building Bazel for
```
Comment (meisterT) [inline third_party/BUILD]: super-nit: please add a . at the end of the comment.

File: third_party/BUILD (hunk 1 above)
Comment (lberki) [reply third_party/BUILD]: It does! It just removes entries from the zip and does not change the time stamps of existing ones. Verified by `unzip -l bazel-genfiles/third_party/netty_tcnative/netty-tcnative-filtered.jar`: timestamps are 2019 April 2. The determinism test also passes.

File: third_party/BUILD (hunk 2 above)
Comment (lberki) [reply third_party/BUILD]: done

**Comment by lberki** [comment]
@buchgr  , is there a way to test that the native code is still being used? I'm not *that* afraid but it would be nice to at least do a manual check.

**Comment by buchgr** [comment]
nice change!

**Comment by buchgr** [comment]
@lberki you could add a unit test trying to load the native code?

**Comment by lberki** [comment]
@buchgr , how do I do that without breaking too many layers of abstraction?

**Comment by buchgr** [comment]
@lberki https://netty.io/wiki/forked-tomcat-native.html#how-to-use ... add a test to the third party package that creates an openssl SslContext ... this should fail if the native libs can't be loaded.

**Comment by lberki** [comment]
@buchgr , I was hoping you'd say that this all is magically tested and I don't need to do anything particular.

Alas, laziness is not the True Way now. PTAL!

**Comment by meisterT** [comment]
Cool, thanks for adding the test and reducing bazel's size!

**Comment by lberki** [comment]
My pleasure :) (actually, it would have been necessary, but a little bit of programming is nice!)

File: src/test/java/com/google/devtools/build/lib/remote/NativeSslTest.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class NativeSslTest {
+  @Test
+  public void nativeSslPresent() throws Exception {
+    SslContextBuilder.forClient().sslProvider(SslProvider.OPENSSL).build();
+  }
```
Comment (buchgr) [inline src/test/java/com/google/devtools/build/lib/remote/NativeSslTest.java]: I hope you don't mind me asking but does this actually fail if you mess up the dynamic libraries stripping?

Comment (lberki) [reply src/test/java/com/google/devtools/build/lib/remote/NativeSslTest.java]: Let me answer in two different ways:

* **Immature:** What do you think I am, an AMATEUR? 
* **Mature:** Actually, I *almost* sent this change out without testing it, but yes, it does --  I verified it by removing `.so` files on Linux from the jar, too. Thanks for keeping me honest :)

**Comment by lberki** [comment]
Ouch. But now, of course, this change doesn't fly because has parts in third_party and out of it. Let me split it, grump.

**Comment by lberki** [comment]
Okay, removed the test and added sent out an internal review. So much ceremony...

**Comment by philwo** [comment]
Pushing this now. Sorry for the delay.

**Comment by philwo** [comment]
Umm... did you already merge this? If yes, please close the PR. :|

**Pull request closed without merging**